### PR TITLE
Add v2/header

### PR DESCRIPTION
**Pull request by jcorbin**

### CODE REVIEW

File: test/v2/header.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    0x65, 0x6e,             // ...
+    0x04, 0x62, 0x6c, 0x75, // hk~1 "blue"
+    0x65,                   // ...
+    0x06, 0x79, 0x65, 0x6c, // hv~1 "yellow"
+    0x6c, 0x6f, 0x77,       // ...
+    0x07, 0x6d, 0x61, 0x67, // hk~1 "magenta"
+    0x65, 0x6e, 0x74, 0x61, // ...
+    0x04, 0x63, 0x79, 0x61, // hv~1 "cyan"
+    0x6e                    // ...
+]);
```
Comment (kriskowal) [inline test/v2/header.js]: Shouldn’t the order be RGB CMY or a cycle of RYGCBM or ROYGBIV? :wink:

File: test/v2/header.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+var test = require('tape');
+var testRead = require('../lib/read_test.js');
+var Header = require('../../v2/header.js');
+
+var testEmpty = Buffer([
+    0x00 // nh:1
+]);
+
+var testSingle = Buffer([
```
Comment (Raynos) [inline test/v2/header.js]: consider adding tests for too little / many key value pairs

File: test/v2/header.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    0x02,                   // nh:1
+    0x03, 0x6b, 0x65, 0x79, // hk~1 "key"
+    0x03, 0x76, 0x61, 0x6c, // hv~1 "val"
+    0x03, 0x6b, 0x65, 0x79, // hk~1 "key"
+    0x03, 0x56, 0x41, 0x4c  // hv~1 "VAL"
+]);
+
+var testNullKey = Buffer([
+    0x01,                   // nh:1
+    0x00, 0x02, 0x65, 0x6b, // hk~1 hv~1 "" => "ek"
```
Comment (Raynos) [inline test/v2/header.js]: this is easier to read as 

```js
[
  0x01
  0x00
  0x02, 0x65, 0x6b
]
```

File: test/v2/header.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+]);
+
+var testNullVal = Buffer([
```
Comment (Raynos) [inline test/v2/header.js]: this is easier to read as

```js
[
  0x01
  0x02, 0x65, 0x76,
  0x00
]
```

**Comment by Raynos** [comment]
lgtm.

**Comment by kriskowal** [comment]
Looks good to me. :ok_hand:

**Pull request merged**